src/sonic_ax_impl/mibs/ietf/rfc2737.py

```diff
@@ -648,6 +674,21 @@ def __init__(self, mib_updater):
         self.entity_to_oid_map = {}
 
     def reinit_data(self):
+
+        # Redis subscriptions are established and consumed in update_data,
+        # but if there's stable exception during update logic,
+        # the reinit_data will be called, but the update_data is never called.
+        # The message is sent into subscription queue, but never got consumed,
+        # this causes Redis memory leaking.
+        # Hence clear the message in the subscription and cancel the subscription during reinit_data
+        for db_index in list(self.pub_sub_dict):
+            pubsub = self.pub_sub_dict[db_index]
+            db_conn = self.mib_updater.statedb[db_index]
+            # clear message in the subscription and cancel the subscription
+            mibs.clear_pubsub_msg(pubsub)
```

Comment:
Seems like the pub_sub_dict is getting cleared upon every reinit_data() call, is that the intention or should the clearing be done only upon exception?

Reply:
Hi @SuvarnaMeenakshi , yes, I intentionally design to clear the subscription on every reinit_data() call. Combined with the external exception catch, we can confirm the subscription will always be cleared.

As for the logic that only clear subscriptions on the exceptions, definitely it makes sense, but there's trade off
if we **keep current logic**
### Adv:
####  We can confirm no memory leaking is expected, because every reinit_data clears the subscriptions.
### Disadv:
####  We re-establish the psubscription to redis server every 60s.

After some research, the psubscription establishment is not heavy and generally considered as light weight.
The heavy works are the message sending after psub established.

If we **only clear on exception**
### Adv:
#### We don't need to re-establish the psubscriptions.
### Disadv:
#### Only when exceptions happened we clear the psubscriptions. If there's no exception raised, the psubscriptions will not be cleared.

From my understanding, semantically, the cleaning of the psub is not a directly result of catching an exception, it's like a patched bug fix saying we know if exception got raised, there is **possibility** that the data in the psub doesn't get consumed(But it could be consumed actually before the exception), so I clean it no matter what the exception is.

But in current approach, I prefer to consider reinit_data a scheduled total refresh, like what it does to the entity caches, clean all of them and fetch from the database no matter got notified by the psub or not. So in the total refresh, we clean the subs, don't sounds like a patched bug fix but a designed feature.
Definitely it would be better if there's a hook specifically to re-initialized all the thing. 
But the reinit_connection only happens on RuntimeError, and I don't want to add new hooks which is a big change to the structure.

@SuvarnaMeenakshi kindly let me know your suggestions and considerations, I will ok with both approaches.

Reply:
> Hi @SuvarnaMeenakshi , yes, I intentionally design to clear the subscription on every reinit_data() call. Combined with the external exception catch, we can confirm the subscription will always be cleared.
> 
> As for the logic that only clear subscriptions on the exceptions, definitely it makes sense, but there's trade off if we **keep current logic**
> 
> ### Adv:
> #### We can confirm no memory leaking is expected, because every reinit_data clears the subscriptions.
> ### Disadv:
> #### We re-establish the psubscription to redis server every 60s.
> After some research, the psubscription establishment is not heavy and generally considered as light weight. The heavy works are the message sending after psub established.
> 
> If we **only clear on exception**
> 
> ### Adv:
> #### We don't need to re-establish the psubscriptions.
> ### Disadv:
> #### Only when exceptions happened we clear the psubscriptions. If there's no exception raised, the psubscriptions will not be cleared.
> From my understanding, semantically, the cleaning of the psub is not a directly result of catching an exception, it's like a patched bug fix saying we know if exception got raised, there is **possibility** that the data in the psub doesn't get consumed(But it could be consumed actually before the exception), so I clean it no matter what the exception is.
> 
> But in current approach, I prefer to consider reinit_data a scheduled total refresh, like what it does to the entity caches, clean all of them and fetch from the database no matter got notified by the psub or not. So in the total refresh, we clean the subs, don't sounds like a patched bug fix but a designed feature. Definitely it would be better if there's a hook specifically to re-initialized all the thing. But the reinit_connection only happens on RuntimeError, and I don't want to add new hooks which is a big change to the structure.
> 
> @SuvarnaMeenakshi kindly let me know your suggestions and considerations, I will ok with both approaches.

Thank you for the details, I understand the approach and sounds like a good way since it is not a heavy operation.